components/cam/src/dynamics/se/dyn_grid.F90

```diff
@@ -76,167 +99,176 @@ module dyn_grid
   public :: dyn_grid_get_colndx
   public :: physgrid_copy_attributes_d
 
-  integer(kind=iMap), pointer :: fdofP_local(:,:) => null()
+  public :: compute_global_area
+  public :: compute_global_coords
 
-  real(r8), public, pointer :: w(:) => null()        ! weights
+  integer(kind=iMap), pointer :: fdofP_local(:,:) => null()
 
-  !! Local lat/lon arrays
-  real(r8), pointer :: pelat_deg(:) => null()  ! pe-local latitudes (degrees)
-  real(r8), pointer :: pelon_deg(:) => null()  ! pe-local longitudes (degrees)
-  real(r8), pointer :: pearea(:) => null()  ! pe-local areas
-  integer(iMap), pointer :: pemap(:) => null()  ! pe-local map for PIO decomp
+  real(r8), public, pointer :: w(:) => null()       ! weights
 
+  ! Local lat/lon arrays
+  real(r8),      allocatable :: pelat_deg(:)        ! pe-local latitudes (degrees)
+  real(r8),      allocatable :: pelon_deg(:)        ! pe-local longitudes (degrees)
+  real(r8),          pointer :: pearea(:) => null() ! pe-local areas
+  integer(iMap),     pointer :: pemap(:)  => null() ! pe-local map for PIO decomp
 
-!========================================================================
+!===================================================================================================
 contains
-!========================================================================
+!===================================================================================================
 
   subroutine dyn_grid_init()
+    use derivative_mod,          only: allocate_subcell_integration_matrix
+    !----------------------------Local-Variables--------------------------------
+    integer :: ie
+    !---------------------------------------------------------------------------
+    if (fv_nphys > 0) then
 
-  end subroutine dyn_grid_init
+      ! Allocate stuff
+      allocate(fv_physgrid(nelemd))
+      do ie = 1,nelemd
+        allocate( fv_physgrid(ie)%lat        (fv_nphys,fv_nphys) )
+        allocate( fv_physgrid(ie)%lon        (fv_nphys,fv_nphys) )
+        allocate( fv_physgrid(ie)%area       (fv_nphys,fv_nphys) )
+        allocate( fv_physgrid(ie)%corner_lat(fv_nphys,fv_nphys,4) )
+        allocate( fv_physgrid(ie)%corner_lon(fv_nphys,fv_nphys,4) )
+      end do ! ie
+
+      ! setup grid coordinates and area
+      call fv_physgrid_init()
 
-  !========================================================================
+      ! allocate subcell integration matrix
+      call allocate_subcell_integration_matrix(np,fv_nphys)
+
+    end if ! fv_nphys > 0
+
+  end subroutine dyn_grid_init
+  !
+  !=================================================================================================
+  !
+  subroutine dyn_grid_final()
```

Comment:
see my comment above in "dyn_comp"

Reply:
This will be in a commit coming here in a couple minutes